tests/openssl_sgx_tcrypto_compat/host/CSchema_checker.cpp

```diff
@@ -0,0 +1,358 @@
+    if (schema)
+    {
+        m_schema =
+            (t_openssl_schema*)new (std::nothrow) t_openssl_schema[schema_size];
+        if (!m_schema)
+            throw "Error allocating schema in CSchemaChecker constructor";
+
+        for (uint i = 0; i < schema_size; i++)
+            m_schema[i] = schema[i];
+    }
+    else
+        m_schema = nullptr;
+}
+CSchemaChecker::~CSchemaChecker()
+{
+    CleanUpParams();
+
+    if (m_schema)
+        delete[] m_schema;
+}
+
+void CSchemaChecker::randomize_api_param(openssl_api_param* p)
+{
+    // for 1 api - randomize params
+    uint64_t type;
+    size_t len;
+    char* buffer;
+    if (!m_schema)
+        return;
+
+    t_openssl_schema* api_schema = &m_schema[m_schema_idx];
+    for (int i = 0; i < OPENSSL_MAX_PARAMETER_COUNT; i++)
+    {
+        buffer = p->p[i];
+        type = api_schema->type[i];
+        if (!SSL_VALID(type))
+            break;
+
+        if (SSL_FIXLEN(type))
+        {
+            len = api_schema->length[i];
+            for (size_t j = 0; j < len; j++)
+                buffer[j] = (char)(rand());
+            OverideRandomizedValue(
+                m_api_id, (uint)i, type, 1, (unsigned char*)buffer, (uint)len);
+        }
+        else if (SSL_VARLEN_X(type))
+        {
+            len = (size_t)m_varlen_values[SSL_WHICH_VARLEN(type)];
+            for (size_t j = 0; j < len; j++)
+                buffer[j] = (char)(rand());
+            OverideRandomizedValue(
+                m_api_id, (uint)i, type, 1, (unsigned char*)buffer, (uint)len);
+        }
+    }
+}
+
+void CSchemaChecker::copy_api_param(
+    openssl_api_param* p2,
+    openssl_api_param* p1)
+{
+    uint64_t type;
+    size_t len;
+
+    if (!m_schema)
+        return;
+    t_openssl_schema* api_schema = &m_schema[m_schema_idx];
+
+    assert(p1->id == p2->id);
+    for (int i = 0; i < OPENSSL_MAX_PARAMETER_COUNT; i++)
+    {
+        char* b1 = p1->p[i];
+        char* b2 = p2->p[i];
+
+        type = api_schema->type[i];
+        if (!SSL_VALID(type))
+            break;
+
+        if (SSL_FIXLEN(type))
+        {
+            len = api_schema->length[i];
+
+            p2->p[i] = new (std::nothrow) char[len];
+            b2 = p2->p[i];
+            assert(b2);
+            memcpy(b2, b1, len);
+        }
+        else if (SSL_VARLEN_X(type))
+        {
+            len = (size_t)m_varlen_values[SSL_WHICH_VARLEN(type)];
+            p2->p[i] = new (std::nothrow) char[len];
+            assert(NULL != p2->p[i]);
+            b2 = p2->p[i];
+            memcpy(b2, b1, len);
+        }
+        else if (SSL_LEN_X(type))
+        {
+            p2->p[i] = p1->p[i];
+        }
+    }
+}
+
+int CSchemaChecker::check_openssl(openssl_api_param* p1, openssl_api_param* p2)
+{
+    assert(p1->id == p2->id);
+
+    return 0;
+}
+
+int CSchemaChecker::SetupParams(openssl_api_id id, uint schema_id)
+{
+    // per 1 api
+    m_api_id = id;
+    m_p1.id = m_p2.id = id;
+    m_schema_idx = schema_id;
```

Comment:
```suggestion
    m_schema_id = schema_id;
```
(or m_schema_index, but either way make the param name and the member name consistent)

Reply:
changed to m_schema_index